src/plugins/controls/public/options_list/components/options_list_popover_invalid_selections.tsx

```diff
@@ -91,8 +92,13 @@ export const OptionsListPopoverInvalidSelections = () => {
         options={selectableOptions}
         listProps={{ onFocusBadge: false, isVirtualized: false }}
         onChange={(newSuggestions, _, changedOption) => {
+          if (!fieldSpec || !changedOption.key) {
+            throw new Error(OptionsListStrings.popover.getInvalidSelectionMessage());
```

Comment:
Where is this error caught when thrown? I think you will need to handle the error in this component and maybe display the error. message in a toast.

Reply:
This actually shouldn't ever be hit - but it is technically being caught by the `EuiErrorBoundary`. I'm not sure if it's worth handling this more elegantly since we can handle this better with the new system? (considering again, this error was thrown as a type safeguard and should never happen)

Reply:
Switched to early return in c9672eedee5c99962f734984936a1a9da0001d6a

Reply:
This is what the rendering would look like if the error is thrown in the callback.

<img width="500" alt="Screenshot 2024-07-26 at 1 50 08 PM" src="https://github.com/user-attachments/assets/810fb13c-ae6c-44eb-9f80-aeb5604c8876">

At a minimum, the callback should not throw and console log a warning and return early. I don't think the argument of "this will never occur" holds water. 1) things change and the error may be triggered by future changes. 2) it provide a poor example that may be duplicated in other locations.

Reply:
Yeah that's fair - I think the error thrown was overkill :+1: Switched to early return in https://github.com/elastic/kibana/commit/c9672eedee5c99962f734984936a1a9da0001d6a